[build] Remove undocumented prefix install vars, use DESTDIR (#14258)
coq/coq: dev/doc/INSTALL.make.md

----------------------------------------

    $ ./configure
    $ make world
    $ make install (you may need superuser rights)

Detailed Installation Procedure.
--------------------------------

> Note this installation instructions are meant for users, to develop
> Coq itself, there is an extra set of targets better suited to that
> purpose, please check the [contributing guide](../../CONTRIBUTING.md).

Reviewer: ```suggestion
These installation instructions are meant for users.  For developers,
there is an extra set of targets better suited to them:
please see the [contributing guide](../../CONTRIBUTING.md).
```

     c.f. https://caml.inria.fr/mantis/view.php?id=7630

   If you want your build to be reproducible, ensure that the
   `SOURCE_DATE_EPOCH` environment variable is set as documented in
   https://reproducible-builds.org/specs/source-date-epoch/

4. Still in the Coq sources directory, do:

        make world

>    to compile Coq (this will default to both native and bytecode
>    unless only the OCaml native compiler is available).

Reviewer: ```suggestion
   to compile Coq (this builds both native and bytecode versions by default
   if both the OCaml native and bytecode compilers are available).
```

     c.f. https://caml.inria.fr/mantis/view.php?id=7630

   If you want your build to be reproducible, ensure that the
   `SOURCE_DATE_EPOCH` environment variable is set as documented in
   https://reproducible-builds.org/specs/source-date-epoch/

4. Still in the Coq sources directory, do:

>         make world

Reviewer 2: Is there any reason to suggest `make world` rather than `make`?

    coqtop          The Coq toplevel
    coqc            The Coq compiler

> Under architecture where `ocamlopt` is available, `coqtop` is the
> native code version of Coq. The byte-code version is named
> `coqtop.byte`, which can be used for debugging purposes. If native
> code isn't available, `coqtop` will point to `coqtop.byte`. `coqc`
> follows a similar scheme.

Reviewer: ```suggestion
For architectures where `ocamlopt` is available, `coqtop` is the
native code version of Coq. The byte-code version is
`coqtop.byte`, which can be used for debugging. If native
code isn't available, `coqtop` will point to `coqtop.byte`. `coqc`
follows a similar scheme.
```

  directory, or read online on http://coq.inria.fr/doc/)
  and in the corresponding manual pages.

Moving Binaries Or Library.
---------------------------

> If you move both the binaries and the library in a consistent way, Coq
> should be able to still run. Otherwise, Coq may not be able to find
> the required prelude files and thenreturn an error message of the
> form:

Reviewer: ```suggestion
If you move both the binaries and the library in a consistent way, Coq
should still be able to run. Otherwise, Coq may not be able to find
the required prelude files and will give this error message:
```
